src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs

```diff
@@ -257,6 +257,18 @@ internal static OtlpMetrics.Metric ToOtlpMetric(this Metric metric)
                             dataPoint.Count = (ulong)metricPoint.GetHistogramCount();
                             dataPoint.Sum = metricPoint.GetHistogramSum();
 
+                            var max = metricPoint.GetHistogramMax();
+                            if (!double.IsNegativeInfinity(max))
```

Comment:
double.IsNaN instead?

Reply:
Would that work? Reading [this](https://learn.microsoft.com/en-us/dotnet/api/system.double.isnan?view=net-7.0#remarks)

> [IsNaN](https://learn.microsoft.com/en-us/dotnet/api/system.double.isnan?view=net-7.0) returns false if a [Double](https://learn.microsoft.com/en-us/dotnet/api/system.double?view=net-7.0) value is either [PositiveInfinity](https://learn.microsoft.com/en-us/dotnet/api/system.double.positiveinfinity?view=net-7.0) or [NegativeInfinity](https://learn.microsoft.com/en-us/dotnet/api/system.double.negativeinfinity?view=net-7.0). To test for these values, use the [IsInfinity](https://learn.microsoft.com/en-us/dotnet/api/system.double.isinfinity?view=net-7.0), [IsPositiveInfinity](https://learn.microsoft.com/en-us/dotnet/api/system.double.ispositiveinfinity?view=net-7.0), and [IsNegativeInfinity](https://learn.microsoft.com/en-us/dotnet/api/system.double.isnegativeinfinity?view=net-7.0) methods.

Reply:
Though, I will admit that `!double.IsNegativeInfinity(max)` does hurt my 🧠 just a little. Maybe we should push the implementation details of negative/positive infinity meaning "unset" somewhere else.

Maybe something like

```csharp
var maxIsSet = metricPoint.TryGetHistogramMax(out var max)
```

Reply:
Another thought... would it ever be the case where max is set but min is not? Maybe it would be nicer for an exporter author to be able to write something like


```csharp
if (metricPoint.HasMinMax()) // This new method does the negative/positive infinity check
{
    dataPoint.Min = metricPoint.GetHistogramMin();
    dataPoint.Max = metricPoint.GetHistogramMax();
}
```

Reply:
I liked the `TryGetHistogramMax` way so switched it to that.

Reply:
**would it ever be the case where max is set but min is not**

^ I cannot think of such case. I like the HasMinMax() slightly better. TryGetHistogramMax/Min() is okay as well. Will unblock PR, and we can adjust if needed

Reply:
I think I will go with the HasMinMax way now after evaluating more carefully. It also seems like this is what Java did. It is complex to use just the current value to determine if a Min/Max is present :P

Reply:
> Is recording an infinity value something we should support?

[The spec address this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#handle-all-normal-values):

> Implementations SHOULD NOT incorporate non-normal values (i.e., +Inf, -Inf, and NaNs) into the sum, min, and max fields, because these values do not map into a valid bucket.